python_modules/libraries/dagster-dg/dagster_dg/cli/plus/deploy_session.py

```diff
@@ -185,19 +194,79 @@ def build_artifact(
     statedir: str,
     use_editable_dagster: bool,
     python_version: Optional[str],
+    location_names: list[str],
 ):
     if not python_version:
         python_version = f"3.{sys.version_info.minor}"
 
-    dockerfile_path = dg_context.root_path / "Dockerfile"
+    requested_location_names = set(location_names)
+
+    if dg_context.is_project:
+        _build_artifact_for_project(
+            dg_context,
+            agent_type,
+            statedir,
+            use_editable_dagster,
+            python_version,
+            workspace_context=None,
+        )
+    else:
+        for spec in dg_context.project_specs:
+            project_root = dg_context.root_path / spec.path
+            project_context: DgContext = dg_context.with_root_path(project_root)
+
+            if (
+                requested_location_names
+                and project_context.code_location_name not in requested_location_names
+            ):
+                continue
+
+            click.echo(f"Building for location {project_context.code_location_name}.")
+            _build_artifact_for_project(
+                project_context,
+                agent_type,
+                statedir,
+                use_editable_dagster,
+                python_version,
+                workspace_context=dg_context,
+            )
+
+
+def _build_artifact_for_project(
+    dg_context: DgContext,
+    agent_type: DgPlusAgentType,
+    statedir: str,
+    use_editable_dagster: bool,
+    python_version: Optional[str],
+    workspace_context: Optional[DgContext],
+):
+    merged_build_config: DgRawBuildConfig = merge_build_configs(
+        workspace_context.build_config if workspace_context else None,
+        dg_context.build_config,
+    )
+
+    build_directory = dg_context.root_path
+    if merged_build_config.get("directory"):
+        build_directory = Path(check.not_none(merged_build_config["directory"]))
+        assert build_directory.is_absolute(), "Build directory must be an absolute path"
```

Comment:
I wonder if we want to have some sort of standard syntax for "relative path to the project root". Wasn't there some discussion in components-land about using like `@/my_path/asdf` to signify what the relative root is?

For now using "relative to the build file" seems totally fine